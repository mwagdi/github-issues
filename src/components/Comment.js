import React from 'react';

import { getDateFormat } from '../helpers';

const Comment = ({comment}) => (
    <div className="comment flex-container justify-space-between">
        <div
        style={{ backgroundImage: `url(${comment.user.avatar_url})`}}
        className="comment__avatar"></div>
        <div className="comment__body">
            <div className="comment__header">
                <span className="comment__login bold">{comment.user.login}</span> commented on {getDateFormat(comment.created_at)}
            </div>
            <div className="comment__content">
                {comment.body}
            </div>
        </div>
    </div>
);

export default Comment;