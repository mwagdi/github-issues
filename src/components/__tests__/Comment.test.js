import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Comment from '../Comment';

test('Comment renders correctly', () => {
    const comment = {
        user: {
            avatar_url: "url"
        }
    }
    const wrapper = shallow(<Comment comment={comment} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
})
