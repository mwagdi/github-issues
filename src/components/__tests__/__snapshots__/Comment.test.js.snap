// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment renders correctly 1`] = `
<div
  className="comment flex-container justify-space-between"
>
  <div
    className="comment__avatar"
    style={
      Object {
        "backgroundImage": "url(url)",
      }
    }
  />
  <div
    className="comment__body"
  >
    <div
      className="comment__header"
    >
      <span
        className="comment__login bold"
      />
       commented on 
      undefined day, NaN
    </div>
    <div
      className="comment__content"
    />
  </div>
</div>
`;
